python/hyperon/atoms.py

```diff
@@ -302,39 +468,40 @@ def clone(self):
         return Bindings(hp.bindings_clone(self.cbindings))
 
     def merge(self, other: 'Bindings') -> 'BindingsSet':
-        """Merges two Bindings frames together into a Bindings Set."""
+        """Merges with another Bindings instance, into a Bindings Set."""
         return BindingsSet(hp.bindings_merge(self.cbindings, other.cbindings))
 
     def add_var_binding(self, var: Union[str, Atom], atom: Atom) -> bool:
-        """Adds a new variable <-> atom association within a bindings"""
+        """Adds a binding between a variable and an Atom."""
         if isinstance(var, Atom):
             return hp.bindings_add_var_binding(self.cbindings, var.get_name(), atom.catom)
         else:
             return hp.bindings_add_var_binding(self.cbindings, var, atom.catom)
 
     def is_empty(self) -> bool:
-        """Checks if a bindings contains no associations. """
+        """Checks if a bindings contains no associations."""
         return hp.bindings_is_empty(self.cbindings)
 
     def narrow_vars(self, vars ):
-        """Removes all variable associations from a bindings except those in the
-        supplied list."""
+        """Keeps only specific variable associations."""
         cvars = hp.CVecAtom = hp.atom_vec_new()
         for var in vars:
             hp.atom_vec_push(cvars, var.catom)
         hp.bindings_narrow_vars(self.cbindings, cvars)
         hp.atom_vec_free(cvars)
 
     def resolve(self, var_name: str) -> Union[Atom, None]:
-        """Returns the atom bound to the supplied variable name in the bindings"""
+        """Finds the atom for a given variable name"""
         raw_atom = hp.bindings_resolve(self.cbindings, var_name)
         return None if raw_atom is None else Atom._from_catom(raw_atom)
 
     def resolve_and_remove(self, var_name: str) -> Union[Atom, None]:
+        """Finds anr removes the atom for a given variable name"""
         raw_atom = hp.bindings_resolve_and_remove(self.cbindings, var_name)
         return None if raw_atom is None else Atom._from_catom(raw_atom)
 
     def iterator(self):
+        """Iterates through all variable-atom pairs in the bindings"""
```

Comment:
While it iterates through all the pairs internally, it returns an iterator, so it doesn't iterate through all of them for an "external observer".